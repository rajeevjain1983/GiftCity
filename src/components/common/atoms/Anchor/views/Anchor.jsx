// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import {
  buildUrl,
  getAsPathWithSlug,
  getMappedPageHref,
  triggerPostMessage,
  getAPIConfig,
} from '../../../../../utils';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

const getAsLinkPath = (IsSlugPathAdded, asPath, incomingUrl) => {
  return IsSlugPathAdded ? asPath || incomingUrl : getAsPathWithSlug(asPath || incomingUrl);
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  asPath,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  url,
  text,
  dataLocator,
  IsSlugPathAdded,
  ...other
}) => {
  const targetVal = target || '_self';
  let incomingUrl = to || url;
  const asLinkPath = getAsLinkPath(IsSlugPathAdded, asPath, incomingUrl);
  if (!noLink) {
    incomingUrl = getMappedPageHref(incomingUrl);
  }
  const hrefUrl = asLinkPath || buildUrl(incomingUrl);

  const apiConfigObj = getAPIConfig();

  let AnchorComponent = null;
  if (children || text) {
    AnchorComponent = noLink ? (
      <a
        href={buildUrl(incomingUrl)}
        className={className}
        onClick={handleLinkClick}
        title={title}
        target={targetVal}
        data-locator={dataLocator}
        {...other}
      >
        {children}
      </a>
    ) : (
      <Link href={incomingUrl} as={asLinkPath} shallow={shallow} scroll={scroll}>
        <a
          className={className}
          href={hrefUrl}
          title={title}
          target={targetVal}
          {...(apiConfigObj.isAppChannel
            ? {
                onClick: () => triggerPostMessage(to || asLinkPath),
              }
            : '')}
          data-locator={dataLocator}
          {...other}
        >
          {children || text}
        </a>
      </Link>
    );
  }
  return AnchorComponent;
};

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  asPath: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.string,
  dataLocator: PropTypes.string,
  IsSlugPathAdded: PropTypes.bool,
};

Anchor.defaultProps = {
  asPath: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
  url: '',
  text: '',
  dataLocator: '',
  IsSlugPathAdded: false,
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
