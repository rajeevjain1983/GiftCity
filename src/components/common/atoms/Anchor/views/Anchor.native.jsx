// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import {
  setTestId,
  getLocator,
  configureInternalNavigationFromCMSUrl,
  isWebViewPage,
} from '@tcp/core/src/utils';
import { StyledText } from '../../../../../../styles/globalStyles/StyledText.style';
import {
  UrlHandler,
  navigateToPage,
  validateExternalUrl,
  redirectToInAppView,
} from '../../../../../utils/index.native';
import withStyles from '../../../hoc/withStyles.native';
import { AnchorStyles, AnchorView, AnchorIcon } from '../Anchor.style.native';
import AccessibilityRoles from '../../../../../constants/AccessibilityRoles.constant';

const Icon = require('../../../../../../../mobileapp/src/assets/images/carrot-small-rights.png');

const getOverrideStyle = overrideStyle => {
  return overrideStyle && overrideStyle.color ? { style: overrideStyle } : null;
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is a Anchor component to manage the internal or external .
 */
const Anchor = ({
  anchorVariation,
  text,
  visible,
  children,
  customStyle,
  locator,
  onPress,
  accessibilityLabel,
  justifyContent,
  hitSlop,
  overrideStyle = {},
  ...otherProps
}) => {
  const { url, navigation, onReuqestClose } = otherProps;
  const accessibilityRole = validateExternalUrl(url)
    ? AccessibilityRoles.Link
    : AccessibilityRoles.Button;
  const openUrl = () => {
    if (isWebViewPage(url) && navigation) {
      if (onReuqestClose) {
        onReuqestClose();
      }
      redirectToInAppView(url, navigation);
    } else if (validateExternalUrl(url)) {
      UrlHandler(url);
    } else if (navigation) {
      const cmsValidatedUrl = configureInternalNavigationFromCMSUrl(url);
      navigateToPage(cmsValidatedUrl, navigation);
    }
  };
  const styleProp = getOverrideStyle(overrideStyle);

  if (children) {
    return (
      <TouchableOpacity
        accessibilityRole={accessibilityRole}
        onPress={onPress || openUrl}
        {...otherProps}
        style={customStyle}
        {...setTestId(getLocator(locator))}
      >
        {children}
      </TouchableOpacity>
    );
  }

  if (text) {
    return (
      <AnchorView
        accessibilityRole={accessibilityRole}
        accessible
        onPress={onPress || openUrl}
        accessibilityLabel={accessibilityLabel || text}
        style={customStyle}
        justifyContent={justifyContent}
        {...setTestId(getLocator(locator))}
        hitSlop={hitSlop}
      >
        <StyledText anchorVariation={anchorVariation} {...otherProps} {...styleProp}>
          {text}
        </StyledText>
        {visible && <AnchorIcon source={Icon} />}
      </AnchorView>
    );
  }

  return null;
};

Anchor.propTypes = {
  anchorVariation: PropTypes.string,
  text: PropTypes.string,
  visible: PropTypes.bool,
  children: PropTypes.shape({}),
  customStyle: PropTypes.shape({}),
  locator: PropTypes.string,
  onPress: PropTypes.func,
  accessibilityLabel: PropTypes.string,
  colorName: PropTypes.string,
  margins: PropTypes.string,
  justifyContent: PropTypes.string,
  hitSlop: PropTypes.shape({}),
  overrideStyle: PropTypes.shape({}),
};

Anchor.defaultProps = {
  anchorVariation: '',
  text: '',
  visible: false,
  children: null,
  customStyle: {},
  locator: '',
  onPress: null,
  accessibilityLabel: '',
  colorName: null,
  margins: null,
  justifyContent: '',
  hitSlop: {},
  overrideStyle: {},
};

export default withStyles(Anchor, AnchorStyles);
export { Anchor as AnchorVanilla };
