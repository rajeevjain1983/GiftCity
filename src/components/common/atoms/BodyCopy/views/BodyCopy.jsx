// 9fbef606107a605d69c0edbcd8029e5d
import React from "react";
import { PropTypes } from "prop-types";
import withStyles from "../../../hoc/withStyle";
import styles from "../BodyCopy.style";

const renderComponent = (compProps) => {
  const {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others
  } = compProps;
  return (
    <Component
      className={className}
      data-locator={dataLocator}
      style={customStyle}
      {...others}
    >
      {children}
    </Component>
  );
};

const BodyCopy = (props) => {
  const {
    children,
    className,
    fontFamily,
    component: Component,
    fontSize,
    fontWeight,
    lineHeight,
    letterSpacing,
    textAlign,
    color,
    dataLocator,
    link,
    customStyle,
    ...others
  } = props;

  const compProps = {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others,
  };
  if (!link) {
    return renderComponent(compProps);
  }
};

const fontFamilyPropTypes = PropTypes.oneOf(["primary", "secondary"]);
const fontWeightPropTypes = PropTypes.oneOf([
  "regular",
  "semibold",
  "extrabold",
  "black",
]);
const letterSpacingPropTypes = PropTypes.oneOf([
  "ls271",
  "ls257",
  "ls222",
  "ls167",
  "normal",
]);
const textAlignPropTypes = PropTypes.oneOf(["left", "center", "right"]);
const colorPropTypes = PropTypes.oneOf([
  "text.primary",
  "text.secondary",
  "text.hint",
  "text.disabled",
  "white",
]);
const fontSizesPropTypes = PropTypes.oneOf([
  "fs48",
  "fs42",
  "fs36",
  "fs32",
  "fs28",
  "fs24",
  "fs22",
  "fs18",
  "fs16",
  "fs15",
  "fs14",
  "fs12",
  "fs10",
  "fs9",
]);

BodyCopy.defaultProps = {
  component: "p",
  className: null,
  children: null,
  fontFamily: "primary",
  fontSize: 12,
  fontWeight: "regular",
  lineHeight: "normal",
  letterSpacing: "normal",
  textAlign: "left",
  color: "text.primary",
  link: null,
  customStyle: {},
};

BodyCopy.propTypes = {
  component: PropTypes.elementType,
  className: PropTypes.string,
  children: PropTypes.node,
  fontFamily: PropTypes.oneOfType([
    PropTypes.arrayOf(fontFamilyPropTypes),
    fontFamilyPropTypes,
  ]),
  fontSize: PropTypes.oneOfType([
    PropTypes.arrayOf(fontSizesPropTypes),
    fontSizesPropTypes,
  ]),
  fontWeight: PropTypes.oneOfType([
    PropTypes.arrayOf(fontWeightPropTypes),
    fontWeightPropTypes,
  ]),
  lineHeight: PropTypes.oneOf(["normal"]),
  letterSpacing: PropTypes.oneOfType([
    PropTypes.arrayOf(letterSpacingPropTypes),
    letterSpacingPropTypes,
  ]),
  textAlign: PropTypes.oneOfType([
    PropTypes.arrayOf(textAlignPropTypes),
    textAlignPropTypes,
  ]),
  color: PropTypes.oneOfType([
    PropTypes.arrayOf(colorPropTypes),
    colorPropTypes,
  ]),
  dataLocator: PropTypes.string,
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    target: PropTypes.string,
    text: PropTypes.string,
  }),
  customStyle: PropTypes.shape({}),
};

const StyledBodyCopy = withStyles(BodyCopy, styles);
StyledBodyCopy.defaultProps = BodyCopy.defaultProps;

export { BodyCopy as BodyCopyVanilla };
export default StyledBodyCopy;
