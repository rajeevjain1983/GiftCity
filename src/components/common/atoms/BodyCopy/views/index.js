// 9fbef606107a605d69c0edbcd8029e5d 
export { default } from './BodyCopy';
