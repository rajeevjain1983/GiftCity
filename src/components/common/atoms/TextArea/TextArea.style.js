// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';

const textareaStyles = css`
  .TextArea__error {
    display: flex;
    flex-direction: row;
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  }
  .warning-icon {
    /* background: transparent url(${getIconPath('circle-alert-fill')}) no-repeat 0 0; */
    background-size: contain;
    border: none;
    height: 14px;
    width: 16px;
    margin-right: 7px;
    flex-shrink: 0;
  }
`;
export default textareaStyles;
