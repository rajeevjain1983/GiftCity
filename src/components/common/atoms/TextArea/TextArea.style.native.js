// 9fbef606107a605d69c0edbcd8029e5d 
import styled from 'styled-components/native';
import { TextInput } from 'react-native';

export const StyledTextInput = styled(TextInput)`
  height: 150px;
  width: 100%;
  border: 1px solid ${props => props.theme.colors.FOOTER.DIVIDER};
`;

export default StyledTextInput;
