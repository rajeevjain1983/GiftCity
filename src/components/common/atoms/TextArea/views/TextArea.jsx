// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';
import StyledTextArea from '../TextArea.style';

export const TextArea = ({
  className,
  id,
  ariaLabel,
  placeholder,
  maxLength,
  inputRef,
  meta,
  dataLocator,
  enableSuccessCheck,
  isRequired,
  errorDataLocator,
  showExplicitError,
  input,
  ...others
}) => {
  const elemValue = input.value;
  const { touched, error } = meta;
  const errorMessagea11yLbl = `textarea__error__${input.name}`;
  const isErrored = (touched && error) || showExplicitError;

  return (
    <label htmlFor={id} className={`${className} input-fields-wrapper`}>
      <textarea
        {...others}
        {...input}
        id={id}
        aria-label={ariaLabel}
        className="TextArea__input"
        name={input.name}
        maxLength={maxLength}
        value={elemValue}
        ref={inputRef}
        placeholder=""
        data-locator={dataLocator}
        aria-required={isRequired}
        aria-describedby={errorMessagea11yLbl}
      />
      <BodyCopy className="TextArea__label" fontFamily="secondary" fontSize="fs12">
        {placeholder}
      </BodyCopy>
      <div className="TextArea__error">
        <div className={isErrored ? 'warning-icon' : ''} aria-disabled="true" />
        <BodyCopy
          color="error"
          component="div"
          fontSize="fs12"
          fontFamily="secondary"
          fontWeight="extrabold"
          role="alert"
          aria-live="assertive"
          data-locator={errorDataLocator}
          id={errorMessagea11yLbl}
        >
          {isErrored ? error || showExplicitError : ''}
        </BodyCopy>
      </div>
      {enableSuccessCheck && <div className="success__checkmark" />}
    </label>
  );
};

TextArea.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.string,
  ariaLabel: PropTypes.string,
  placeholder: PropTypes.string,
  maxLength: PropTypes.number.isRequired,
  inputRef: PropTypes.node.isRequired,
  meta: PropTypes.shape({}),
  dataLocator: PropTypes.string,
  enableSuccessCheck: PropTypes.bool,
  isRequired: PropTypes.bool,
  errorDataLocator: PropTypes.string,
  input: PropTypes.shape({}).isRequired,
  showExplicitError: PropTypes.string,
};

TextArea.defaultProps = {
  id: '',
  ariaLabel: '',
  placeholder: '',
  errorDataLocator: '',
  dataLocator: '',
  meta: {},
  isRequired: false,
  enableSuccessCheck: false,
  showExplicitError: '',
};

export default withStyles(TextArea, StyledTextArea);
