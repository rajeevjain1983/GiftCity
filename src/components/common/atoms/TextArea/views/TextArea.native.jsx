// 9fbef606107a605d69c0edbcd8029e5d 
import React, { useState } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import Image from '../../Image';
import {
  StyledErrorWrapper,
  ViewWithSpacing,
  BodyCopyWithSpacing,
} from '../../styledWrapper/styledWrapper.native';
import { StyledTextInput } from '../TextArea.style.native';

const errorIcon = require('../../../../../../../mobileapp/src/assets/images/alert-triangle.png');

const TextArea = ({
  input,
  id,
  name,
  meta: { error, touched },
  label,
  keyboardType,
  autoCapitalize,
}) => {
  const [focused, setFocused] = useState(false);

  const onFocusHandler = () => {
    setFocused(true);
  };

  const onBlurHandler = () => {
    setFocused(false);
    if (input.onBlur) {
      input.onBlur();
    }
  };

  const onChangeHandler = text => {
    if (input.onChange) {
      input.onChange(text);
    }
  };

  return (
    <View>
      {label && (
        <BodyCopyWithSpacing
          fontFamily="secondary"
          fontSize="fs12"
          fontWeight="regular"
          color="gray.900"
          text={label}
          spacingStyles="margin-bottom-XS"
        />
      )}
      <StyledTextInput
        id={id}
        name={name}
        value={input.value}
        onChangeText={onChangeHandler}
        multiline
        numberOfLines={4}
        onFocus={onFocusHandler}
        onBlur={onBlurHandler}
        onEndEditing={onBlurHandler}
        keyboardType={keyboardType}
        autoCapitalize={autoCapitalize}
        returnKeyType="next"
      />
      {!focused && error && touched && (
        <StyledErrorWrapper>
          <ViewWithSpacing spacingStyles="margin-right-XS">
            <Image source={errorIcon} width="16px" height="14px" />
          </ViewWithSpacing>
          <BodyCopy
            fontFamily="secondary"
            fontWeight="extrabold"
            fontSize="fs12"
            text={error}
            color="error"
          />
        </StyledErrorWrapper>
      )}
    </View>
  );
};

TextArea.propTypes = {
  input: PropTypes.shape({}),
  meta: PropTypes.shape({}),
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  keyboardType: PropTypes.string,
  autoCapitalize: PropTypes.string,
};

TextArea.defaultProps = {
  input: {},
  meta: {},
  label: '',
  keyboardType: 'default',
  autoCapitalize: 'sentences',
};

export default TextArea;
