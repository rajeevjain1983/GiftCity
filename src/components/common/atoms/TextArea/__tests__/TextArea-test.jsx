// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { TextArea } from '../views/TextArea';

describe('Textarea component', () => {
  it('renders correctly', () => {
    const props = {
      id: 'abcd',
      className: 'asdfasdf',
      meta: 'sdfdsf',
      input: { value: 'hello sonar' },
    };
    const component = shallow(<TextArea {...props} />);
    expect(component).toMatchSnapshot();
  });
});
