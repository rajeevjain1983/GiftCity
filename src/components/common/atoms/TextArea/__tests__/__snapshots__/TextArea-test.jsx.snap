// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component renders correctly 1`] = `
<label
  className="asdfasdf input-fields-wrapper"
  htmlFor="abcd"
>
  <textarea
    aria-describedby="textarea__error__undefined"
    aria-label=""
    aria-required={false}
    className="TextArea__input"
    data-locator=""
    id="abcd"
    placeholder=""
    value="hello sonar"
  />
  <Styled(BodyCopy)
    className="TextArea__label"
    color="text.primary"
    component="p"
    customStyle={Object {}}
    fontFamily="secondary"
    fontSize="fs12"
    fontWeight="regular"
    letterSpacing="normal"
    lineHeight="normal"
    link={null}
    textAlign="left"
  />
  <div
    className="TextArea__error"
  >
    <div
      aria-disabled="true"
      className=""
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      data-locator=""
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="textarea__error__undefined"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    />
  </div>
</label>
`;
